### ISSUE TITLE
不支持mysql8.0--json添加索引的语法

### ISSUE THREAD

**Issue by zdm-wyx** [hanchuanchuan/goInception#600]
**描述**
对问题简单清晰的描述
使用goInception-v1.3.0-72，检查mysql8.0-json添加索引时，返回语法错误，但是在mysql8.0中执行时没有问题
**重现**
模拟重现的步骤
<img width="711" alt="image" src="https://github.com/hanchuanchuan/goInception/assets/142984155/c72cc61d-8e6a-401c-8d4b-e01af2470402">

<img width="666" alt="image" src="https://github.com/hanchuanchuan/goInception/assets/142984155/29b90fbc-bc48-4b84-9ed2-39d064a5776d">

**环境**
 - 数据库: [mysql]
 - 版本: [例如 8.0]
 - goInception-v1.3.0-72

**参数**
可能与问题相关的设置参数

**Comment by zdm-wyx** [comment]
@hanchuanchuan

**Comment by qiqizjl** [comment]
一样的问题 看起来好像是解析器部分没有支持 json

**Comment by zdm-wyx** [comment]
我看它那个语句直接去内嵌的tidb执行了，好像tidb不支持这个语法，只能等作者升级go inception了？

**Comment by qiqizjl** [comment]
> 我看它那个语句直接去内嵌的tidb执行了，好像tidb不支持这个语法，只能等作者升级go inception了？

对~ 我们目前的需求还是在 default 对 json 有需求。暂时内部 fork 了一个版本魔改了https://github.com/hanchuanchuan/goInception/blob/master/parser/parser.y 看起来他的语句都是使用 yacc解析的。主要是 tidb 对 json 的支持没有 MySQL 的全面。所以看起来只能自己改了。。

**Comment by zdm-wyx** [comment]
> > 我看它那个语句直接去内嵌的tidb执行了，好像tidb不支持这个语法，只能等作者升级go inception了？
> 
> 对~ 我们目前的需求暂时还是 default 支持 json_array。暂时内部 fork 了一个版本魔改了https://github.com/hanchuanchuan/goInception/blob/master/parser/parser.y 看起来他的语句都是使用 yacc解析的。主要是 tidb 对 json 的支持没有 MySQL 的全面。所以看起来只能自己改了。。

谢谢🙏

**Comment by czxin788** [comment]
现在还不支持，希望作者合并到官方goinception里面 @hanchuanchuan

**Comment by lvai125** [comment]
上面问题和我的类似应该不是json的问题，是索引里使用了函数导致的吧，我的一个语句里包含了这样 (CASE WHEN is_delete = 0 THEN 0 END) )，审核就报错
